js/spellChecker/spellChecker.js

```diff
@@ -0,0 +1,57 @@
+const { remote } = require('electron');
+const { MenuItem } = remote;
+const { isMac } = require('./../utils/misc');
+const { SpellCheckHandler, ContextMenuListener, ContextMenuBuilder } = require('electron-spellchecker');
+
+class SpellCheckHelper {
+
+    constructor() {
+        this.spellCheckHandler = new SpellCheckHandler();
+    }
+
+    /**
+     * Method to initialize spell checker
+     */
+    initializeSpellChecker() {
+        this.spellCheckHandler.attachToInput();
+
+        // This is only for window as in mac the
+        // language is switched w.r.t to the current system language.
+        //
+        // In windows we need to implement RxJS observable
+        // in order to switch language dynamically
+        if (!isMac) {
+            this.spellCheckHandler.switchLanguage('en-US');
```

Comment:
# Windows
Spell checker is only being attached to the `input` HTML tag (which switches the language on the fly.)
Where as in the client app the main input field is custom built which is a `div` element. So setting `en-US` as the starting language for windows.  

 In order to make this dynamic we need to implement `RxJs observable` or any standard event listeners

# macOS
Spell checker is not attached to any `inputs`. The language that is suggested will be the same as the system language.

We can use `RxJs observable` here as well to switch the language dynamically.